stack-graphs/src/lib.rs

```diff
@@ -70,3 +72,25 @@ pub mod stitching;
 pub(crate) mod utils;
 #[cfg(feature = "json")]
 pub mod visualization;
+
+/// Trait to signal that the execution is cancelled
+pub trait CancellationFlag {
+    fn check(&self, at: &'static str) -> Result<(), CancellationError>;
+}
+
+pub struct CancellationFlags;
+impl CancellationFlags {
+    pub fn none() -> impl CancellationFlag {
+        struct NoCancellation;
+        impl CancellationFlag for NoCancellation {
+            fn check(&self, _at: &'static str) -> Result<(), CancellationError> {
+                Ok(())
+            }
+        }
+        NoCancellation
+    }
+}
```

Comment:
ditto https://github.com/tree-sitter/tree-sitter-graph/pull/94#discussion_r949466998: slight preference for exposing `NoCancellation` directly instead of wrapping it in a factory function